specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -385,10 +393,20 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
     }
 
     const partialFilters = Object.fromEntries(
-      Object.entries(filters).map(([fieldName, { value }]) => [
-        fieldName,
-        value,
-      ])
+      Object.entries(filters).map(([fieldName, { value }]) => {
+        const leafField = getFieldsFromPath(
+          this.resource.specifyTable,
+          fieldName
+        ).at(-1);
+          if (
+            rule.isDatabaseConstraint &&
+            typeof value === 'string' &&
+            leafField !== undefined &&
+            leafField.isRelationship === false
+          )
+            return [`${fieldName}${djangoLookupSeparator}iexact`, value];
```

Comment:
(optional) 

To make this a little more readable, we could add `iexact` to the `backendFilter` function: 

https://github.com/specify/specify7/blob/87e1107cb16365abcc910eb98d6f275a49f31f3d/specifyweb/frontend/js_src/lib/components/DataModel/helpers.ts#L122-L142

This could then be modified to something like the following (this could probably be cleaned up even further):

```suggestion
            return Object.entries(backendFilter(fieldName).caseInsensitiveEquals(value)).at(0)!
```